Articles/Blog/2025-02-coding-stanards-for-sustainable-software.md

```diff
@@ -0,0 +1,291 @@
+We aim to develop the initial set of standards at the 2025 S3C in Denver, Colorado.
+
+Having said that, you could focus only on your organization and try idea #3.
+
+**Idea #2: Start measuring technical debt (T1/T)**
+
+This is a great starting point to assess how your team is performing.
+If you are on a Scrum team, this could be done by comparing total sprint capacity (in points) to the number of points in the sprint allocated for developing new capabilities.
+This measure is not meant to be perfect, but it will give you an idea of where you stand.
+Do not be surprised if it is around 50%.
+If your organization uses external support staff to handle user support issues, you can include their capacity as part of the total time T.
+
+
+**Idea #3: Organizations should define minimum quality criteria (Definition of Done)**
+
+Why do so many organizations avoid addressing this issue?
+If your organization has multiple software development projects, should you not strive for consistency in the quality of code being developed and delivered?
+This requires courageous leadership.
+After all, who wants to be the one to tell their software development teams that this list is the minimum quality criteria?
+Consider making a team event to create the initial list.
+Bring change agents from different teams together, create an initial starting point, and then establish a structure that allows these criteria to be updated regularly.
+
+Consider using the S3C conference as a venue for thinking about and discussing these ideas further.
+
+
+### Conclusion
+
+We believe that many organizations can improve their software development teams by focusing on improving knowledge and practices for creating sustainable software.
+While there is not excellent scientific research on this topic, we should not be flying blind.
+We should take a proactive approach to developing software sustainably.
+At one facility, we estimate that if teams could move their practices toward sustainable standards/practices, there could be a savings of $100M annually.
+Yes, that is correct.
+That is how much technical debt can impact an organization, and worse, some might not even know they are swimming in it.
+
+### Acknowledgments
+
+We thank Scott Warnock, Stuart Baxley, Chris Sullivan, Gary Lawson, Akhil Potla, Richard Drake, Henry Gabaldon, Erik Strack, Terri Galpin, Tricia Gharagazloo, Dena Vigil, Richard Michael Jack Kramer, Salomé Thorson, Raisa Koshkin, and the Sierra Toolkit Team (Alan Williams, Jesse Thomas, Todd Coffey, Nate Roehrig, C. Riley Wilson, Tolu Okusanya, David Glaze, and Johnathan Vo) for their invaluable contributions. We also acknowledge Roscoe Bartlett for his valuable suggestions for this article.
+
+
+### Author bio
+
+[Manoj K. Bhardwaj, Ph.D.](https://www.linkedin.com/in/manoj-bhardwaj-22448932/), is a technical staff member at Sandia National Laboratories with a lengthy background in numerical software development and simulation.
```

Comment:
Looks good:

* https://preview.bssw.io/blog_posts/developing-coding-standards-and-practices-for-sustainable-software-development